packages/constants/src/assets.test.ts

```diff
@@ -56,34 +56,79 @@ describe('assetPatterns', () => {
+      expect(assetPatterns.votingReceipt.matches('voted_on_abc123')).toBe(true);
     });
 
     it('does not match when a string contains, but does not begin with, `voted_on_`', () => {
-      expect(assetPatterns.votingReceipt.test('ibc-transfer/channel-1234/voted_on_abc123')).toBe(
+      expect(assetPatterns.votingReceipt.matches('ibc-transfer/channel-1234/voted_on_abc123')).toBe(
         false,
       );
     });
   });
 
   describe('ibc', () => {
     it('matches when a string follows the pattern transfer/<channel>/<denom>', () => {
-      expect(assetPatterns.ibc.test('transfer/channel-141/uosmo')).toBeTruthy();
-      expect(assetPatterns.ibc.test('transfer/channel-0/upenumbra')).toBeTruthy();
-      expect(assetPatterns.ibc.test('transfer/channel-0/upenumbra/moo/test')).toBeTruthy();
-      expect(assetPatterns.ibc.test('x/channel-0/upenumbra')).toBeFalsy();
+      expect(assetPatterns.ibc.matches('transfer/channel-141/uosmo')).toBeTruthy();
+      expect(assetPatterns.ibc.matches('transfer/channel-0/upenumbra')).toBeTruthy();
+      expect(assetPatterns.ibc.matches('transfer/channel-0/upenumbra/moo/test')).toBeTruthy();
+      expect(assetPatterns.ibc.matches('x/channel-0/upenumbra')).toBeFalsy();
     });
 
     it('captures channel and denom correctly', () => {
-      const match = 'transfer/channel-141/uosmo'.match(assetPatterns.ibc);
-      expect(match?.groups?.['channel']).toBe('channel-141');
-      expect(match?.groups?.['denom']).toBe('uosmo');
+      const match = assetPatterns.ibc.capture('transfer/channel-141/uosmo');
+      expect(match?.channel).toBe('channel-141');
+      expect(match?.denom).toBe('uosmo');
     });
 
     it('captures multi-hops', () => {
-      const match = 'transfer/channel-141/channel-42/uosmo'.match(assetPatterns.ibc);
-      expect(match?.groups?.['channel']).toBe('channel-141');
-      expect(match?.groups?.['denom']).toBe('channel-42/uosmo');
+      const match = assetPatterns.ibc.capture('transfer/channel-141/channel-42/uosmo');
+      expect(match?.channel).toBe('channel-141');
+      expect(match?.denom).toBe('channel-42/uosmo');
     });
   });
 });
+
+describe('RegexMatcher', () => {
+  describe('RegexMatcher.matches', () => {
+    it('should return true when the string matches the regex', () => {
+      const regex = /^[a-z]+$/;
+      const matcher = new RegexMatcher(regex);
+      expect(matcher.matches('abc')).toBe(true);
+    });
+
+    it('should return false when the string does not match the regex', () => {
+      const regex = /^[a-z]+$/;
+      const matcher = new RegexMatcher(regex);
+      expect(matcher.matches('123')).toBe(false);
+    });
+  });
+
+  it('should return undefined if no groups are present', () => {
+    const regex = /hello/;
+    const matcher = new RegexMatcher(regex);
+    // eslint-disable-next-line @typescript-eslint/no-confusing-void-expression
+    expect(matcher.capture('hello')).toBeUndefined();
```

Comment:
Kinda nice the type system knows it shouldn't be calling something without capture groups